src/protagonist/API/Features/NamedQueries/NamedQueriesController.cs

```diff
@@ -0,0 +1,166 @@
+    /// </summary>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     POST: /customers/1/namedQueries
+    ///     {
+    ///         "name":"my-named-query",
+    ///         "template":"space=example"
+    ///     }
+    /// </remarks>
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status409Conflict)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> PostNamedQuery(
+        [FromRoute] int customerId,
+        [FromBody] NamedQuery newNamedQuery,
+        [FromServices] HydraNamedQueryValidator validator,
+        CancellationToken cancellationToken)
+    {
+        if (newNamedQuery.Global == true && !User.IsAdmin()) 
+            return this.HydraProblem("Only admins are allowed to create global Named Queries", null, 400);
+        
+        var validationResult = await validator.ValidateAsync(newNamedQuery, cancellationToken);
+        if (!validationResult.IsValid)
+        {
+            return this.ValidationFailed(validationResult);
+        }
+        
+        var request = new CreateNamedQuery(customerId, newNamedQuery.ToDlcsModel(customerId));
+        
+        return await HandleUpsert(request, 
+            nq => nq.ToHydra(GetUrlRoots().BaseUrl),
+            errorTitle: "Failed to update Named Query",
+            cancellationToken: cancellationToken);
+    }
+
+    /// <summary>
+    /// Get a specified Named Query, either global or owned by the user
+    /// </summary>
+    [HttpGet]
+    [Route("{namedQueryId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetNamedQuery(
+        [FromRoute] int customerId,
+        [FromRoute] string namedQueryId,
+        CancellationToken cancellationToken)
+    {
+        var namedQuery = new GetNamedQuery(customerId, namedQueryId);
+        
+        return await HandleFetch(
+            namedQuery,
+            nq => nq.ToHydra(GetUrlRoots().BaseUrl),
+            errorTitle: "Failed to get Named Query",
+            cancellationToken: cancellationToken
+        );
+    }
+    
+    /// <summary>
+    /// Update an existing Named Query owned by the user - currently, only the template can be modified
+    /// </summary>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     PUT: /customers/1/namedQueries/a90d6e44-4cdb-410b-999e-30c2ea3955b2
+    ///     {
+    ///         "template":"space=example-updated"
+    ///     }
+    /// </remarks>
+    [HttpPut]
+    [Route("{namedQueryId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> PutNamedQuery(
+        [FromRoute] int customerId,
+        [FromRoute] string namedQueryId,
+        [FromBody] NamedQuery newNamedQuery,
```

Comment:
typo in `newNamedQuery` as it's an update

Reply:
Changed to namedQueryChanges in 4b1179bb00229e5265b27dff322bc283440cb3d5